### PR TITLE
Add tests for shader program compilation

### PR DESCRIPTION
**Pull request by chrishavlin**
This adds some new tests for checking that the shader programs compile. Had to modify `yt_idv.shader_objects` a bit to make it more straightforward.

### CODE REVIEW

File: yt_idv/tests/test_yt_idv.py [hunk 1]
```diff
@@ -151,10 +152,67 @@ def test_curves(osmesa_fake_amr, image_store):
     curve_collection.add_data()  # call add_data() after done adding curves
 
     cc_render = CurveCollectionRendering(
-        data=curve_collection, curve_rgb=(0.2, 0.2, 0.2, 1.0), line_width=4
+        data=curve_collection, curve_rgba=(0.2, 0.2, 0.2, 1.0), line_width=4
```
Comment (chrishavlin) [inline yt_idv/tests/test_yt_idv.py:155]: unrelated fix here.

File: yt_idv/shader_objects.py [hunk 2]
```diff
@@ -252,6 +253,7 @@ class Shader(traitlets.HasTraits):
     )
     blend_equation = GLValue("func add")
     depth_test = GLValue("always")
+    _filename = traitlets.Unicode()
```
Comment (chrishavlin) [inline yt_idv/shader_objects.py:256]: added this attr to have a convenient way of printing error messages if compiling a shader from a file

File: yt_idv/shader_objects.py [hunk 3]
```diff
@@ -243,6 +243,7 @@ class Shader(traitlets.HasTraits):
     """
 
     _shader = None
+    allow_null = traitlets.Bool(True)
```
Comment (chrishavlin) [inline yt_idv/shader_objects.py:246]: IMO keeping this at True is the way to go so that the current behavior is default -- don't want shader compilation errors to crash the GUI, and debugging or re-compiling new shaders on the fly is nicer when the exception is caught. But adding this attr allows a convenient switch to flip during testing.

Comment (matthewturk) [reply yt_idv/shader_objects.py:251]: I think that's a good point.  We could also expose it per-shader-using object as a traitlet, and then have that be passed down.  That may be overkill though.

File: yt_idv/shader_objects.py (hunk 2 above)
Comment (matthewturk) [reply yt_idv/shader_objects.py:261]: :+1:

**Review by matthewturk — APPROVED** [review]

**Comment by chrishavlin** [comment]
pre-commit.ci run
